### PR TITLE
Support cloud upload for Koji composes

### PR DESCRIPTION
**Pull request by thozza**
This PR changes the Cloud API implementation and relevant worker jobs to allow specifying cloud upload target for Koji composes. Koji composes are always uploaded to Koji (the target is added always implicitly to the image request), however with this change, an additional cloud target can be specified as well.

### Notable changes

- Allow specifying upload options in Cloud API for Koji composes.
- Support `gce-rhui` image type in Cloud API, which was not supported so far.
- Enhance the `koji-finalize` job implementation to handle more than one upload target for koji composes.
- Enhance the `koji.sh` test case to also test Koji composes with cloud upload.
- Test cloud upload with Koji composes in CI for all supported cloud providers and their respective RHUI images.
- The `osbuild` job implementation now contains a workaround to allow uploading XZ compressed images to AWS and Azure. This is done by checking if the image filename is suffixed with `.xz` before uploading it and if it is, the implementation extracts it. The workaround will be needed until multiple exports are supported by image definitions and different exports are used per upload target (`osbuild` job implementation already supports this).

- [x] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: internal/cloudapi/v2/handler.go [hunk 1]
```diff
@@ -627,10 +640,28 @@ func (h *apiHandlers) getComposeStatusImpl(ctx echo.Context, id string) error {
 			if err != nil {
 				return HTTPError(ErrorGettingBuildDependencyStatus)
 			}
+
+			var us *UploadStatus
+			// only single upload target in addition to Koji is allowed
+			if len(buildJobResult.TargetResults) > 2 {
+				return HTTPError(ErrorSeveralUploadTargets)
+			}
```
Comment (gicmo) [inline internal/cloudapi/v2/handler.go:648]: That one of them is actually a koji target is ensured by `enqueueKojiCompose`, i.e. the specialised compose handler we use when `request.Koji` is present. That tripped me up a bit.

File: tools/koji-compose.py [hunk 2]
```diff
@@ -56,40 +56,41 @@ def compose_request(distro, koji, arch):
...
-        print("Failed to create compose")
-        print(r.text)
+        print("Failed to create compose", file=sys.stderr)
+        print(r.text, file=sys.stderr)
         sys.exit(1)
 
-    print(r.text)
+    print(r.text, file=sys.stderr)
     compose_id = r.json()["id"]
+    print(compose_id)
```
Comment (gicmo) [inline tools/koji-compose.py:141]: Alternatively could have written it to a file and then get it from there, but I think it is fine.

**Review by gicmo — APPROVED** [review]
Looks all good to me! Thanks so much!

**Review by ondrejbudai — APPROVED** [review]

File: cmd/osbuild-worker/jobimpl-koji-finalize.go [hunk 3]
```diff
@@ -153,13 +153,15 @@ func (impl *KojiFinalizeJobImpl) Run(job worker.Job) error {
 		// multiple
 		buildRPMs = rpmmd.DeduplicateRPMs(buildRPMs)
 
-		// TODO: support multiple upload targets
-		if len(buildArgs.TargetResults) != 1 {
-			kojiFinalizeJobResult.JobError = clienterrors.WorkerClientError(clienterrors.ErrorKojiFinalize, "Koji compose OSBuild job result must contain exactly one target result")
+		kojiTargetResults := buildArgs.TargetResultsByName(target.TargetNameKoji)
+		// Only a single Koji target is allowed per osbuild job
+		if len(kojiTargetResults) != 1 {
+			kojiFinalizeJobResult.JobError = clienterrors.WorkerClientError(clienterrors.ErrorKojiFinalize, "Exactly one Koji target results are expected per osbuild job")
```
Comment (achilleas-k) [inline cmd/osbuild-worker/jobimpl-koji-finalize.go:159]: Syntax issue: `Exactly one Koji target result is expected per osbuild job`

**Review by achilleas-k — APPROVED** [review]
LGTM.  One minor nitpick below.

File: cmd/osbuild-worker/jobimpl-koji-finalize.go (hunk 3 above)
Comment (teg) [reply cmd/osbuild-worker/jobimpl-koji-finalize.go:159]: address in follow up?

File: tools/koji-compose.py (hunk 2 above)
Comment (thozza) [reply tools/koji-compose.py:141]: I agree. A matter of personal preference I guess 🙂

File: cmd/osbuild-worker/jobimpl-koji-finalize.go (hunk 3 above)
Comment (thozza) [reply cmd/osbuild-worker/jobimpl-koji-finalize.go:159]: Happy to fix in a follow up 😉  since a green CI is a rare commodity...

**Pull request merged**